src/App.vue

```diff
@@ -45,8 +45,18 @@
               </header>
 
               <stepper class="mt-10" />
-
-              <router-view />
+              <!-- Sign in and sign out components -->
+              <sign-in v-if="isRouteName(RouteNames.SIGN_IN)" />
+              <sign-out v-if="isRouteName(RouteNames.SIGN_OUT)" />
+
+              <!-- Only render when data is ready, or validation can't be properly evaluated. -->
+              <template v-if="haveData">
+                <!-- Using v-show to pre-create/mount components so validation on stepper is shown -->
+                <define-company v-show="isRouteName(RouteNames.DEFINE_COMPANY)" />
+                <add-people-and-roles v-show="isRouteName(RouteNames.ADD_PEOPLE_AND_ROLES)" />
+                <create-share-structure v-show="isRouteName(RouteNames.CREATE_SHARE_STRUCTURE)" />
+                <review-confirm v-show="isRouteName(RouteNames.REVIEW_CONFIRM)" />
+              </template>
```

Comment:
- Sign in and sign out components are using v-if as they are not needed all the time. Using v-show instead cause state issues and occasionally a redirect loop when refreshing the screen
- a template is used to wrap the view components to eliminate any timing issues, if mounting occurs too soon and the data is not ready, the validation is not properly evaluated and properly reflected in the stepper

Reply:
We will have to see if there's a way to specify the steps components programmatically so that we can just loop over them to support other corp types in the future. (I know this is possible in Angular.)

Pseudo-code:
```
for (step in steps)
{
  <programmatic-component v-show="is current component" ... />
}
```

Maybe this? https://medium.com/js-dojo/simply-switching-in-between-the-components-dynamically-vue-js-a436fe10008a

Reply:
Some routes are applicable for some entity types only. Example Share structure is only for BCOMP and Rules is only for CP (Yet to be implemented).

Reply:
This would be nice too (idea from my Angular days):
https://github.com/vuejs/vue/issues/8097

(Not mandatory at this time, and might not be usable with "for" loop.)

Reply:
I will take a look! Thanks

Reply:
Updated the components to be rendered based off of the step config.

My thoughts were to update `getSteps()` in the future to return the relevant steps config depending on entity type when it is needed and it will just automatically show what is relevant.

Reply:
Woohoo! Looks great... love it!